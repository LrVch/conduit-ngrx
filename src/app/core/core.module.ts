import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
