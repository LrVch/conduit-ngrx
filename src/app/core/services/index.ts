export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
