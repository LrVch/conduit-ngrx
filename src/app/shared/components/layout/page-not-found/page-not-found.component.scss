.page {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
    top: 0;
    font-family: "Courier new";
    font-size: 18px;
    line-height: 1.5em;
    background: #081421;
    color: #d3d7de;
}

.code-area > span {
    display: block;
}

a {
	color: #fff;
}