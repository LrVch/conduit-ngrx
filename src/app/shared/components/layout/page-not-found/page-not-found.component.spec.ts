import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '../../../lib/testing';

import { PageNotFoundComponent } from './page-not-found.component';
import { DebugElement } from '@angular/core';

describe('PageNotFoundComponent', () => {
    let component: PageNotFoundComponent;
    let fixture: ComponentFixture<PageNotFoundComponent>;
    let de: DebugElement;
    let el: HTMLElement;

    beforeEach(
        async(() => {
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [PageNotFoundComponent]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(PageNotFoundComponent);
                component = fixture.componentInstance;
                de = fixture.debugElement;
                el = de.nativeElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should compile', () => {
        expect(fixture).toMatchSnapshot();
    });
});

