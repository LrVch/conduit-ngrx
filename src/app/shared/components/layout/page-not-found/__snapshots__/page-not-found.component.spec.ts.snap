// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFoundComponent should compile 1`] = `
<app-page-not-found>
  <div
    class="page"
  >
    <div
      class="container"
    >
      <div
        class="main-content"
      >
        <div
          class="row"
        >
          <div
            class="col-sm-12"
          >
            <div
              class="code-area"
            >
              <span
                style="color: #777;font-style:italic;"
              >
                 // 404 page not found. 
              </span>
              <span>
                <span
                  style="color:#d65562;"
                >
                   if 
                </span>
                 (
                <span
                  style="color:#4ca8ef;"
                >
                  !
                </span>
                <span
                  style="font-style: italic;color:#bdbdbd;"
                >
                  found
                </span>
                ) { 
              </span>
              <span>
                <span
                  style="padding-left: 15px;color:#2796ec"
                >
                  <i
                    style="width: 10px;display:inline-block"
                  />
                  throw 
                </span>
                <span>
                   (
                  <span
                    style="color: #a6a61f"
                  >
                    "(╯°□°)╯︵ ┻━┻"
                  </span>
                  ); 
                </span>
                <span
                  style="display:block"
                >
                  }
                </span>
                <span
                  style="color: #777;font-style:italic;"
                >
                  <a
                    routerlink="/"
                  >
                    Go home!
                  </a>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</app-page-not-found>
`;
